pages/docs/reference/type-safe-builders.md

```diff
@@ -2,20 +2,18 @@
 type: doc
 layout: reference
 category: "Syntax"
-title: "Type-Safe Groovy-Style Builders"
+title: "Type-Safe Builders"
 ---
 
 # Type-Safe Builders
 
-The concept of [builders](http://www.groovy-lang.org/dsls.html#_nodebuilder) is rather popular in the *Groovy* community.
-Builders allow for defining data in a semi-declarative way. Builders are good for [generating XML](http://www.groovy-lang.org/processing-xml.html#_creating_xml), 
-[laying out UI components](http://www.groovy-lang.org/swing.html), 
-[describing 3D scenes](http://www.artima.com/weblogs/viewpost.jsp?thread=296081) and more...
+By using well-named functions as builders in combination with extension lambdas it is possible to create static type safe builders in Kotlin. Type-safe builders are also known as domain specific languages (DSLs).
 
-For many use cases, Kotlin allows to *type-check* builders, which makes them even more attractive than the 
-dynamically-typed implementation made in Groovy itself.
+Builders, or domain specific languages, are a good solution for often-repeated code that builds complex data structures in a semi-declarative way. Builders are good for [generating XML](http://www.groovy-lang.org/processing-xml.html#_creating_xml), 
```

Comment:
I like the phrasing of '*complex data structures*'. We can make it even better by adding something like *hierarchical* or *potentially nested*.